### ISSUE TITLE
after image resize, the text is blurry and unreadable

### ISSUE THREAD

**Issue by chaimch** [imgproxy/imgproxy#942]
I've tried many cases in the pro version, hoping to fix the problem that the text doesn't get blurred after resize, but none of them worked as expected.  Our business has a lot of usage scenarios of this kind, and I'm hoping this will help me solve the problem. Here are the notes from my tests.

---

the original image is as follows:
![ppcc](https://user-images.githubusercontent.com/12760278/183043219-9e43221c-c688-4785-8f96-d6e83e9f9b21.jpg)

/%processing_options/ -> /resize:auto:1024:0/gravity:sm/enlarge:0/quality:100/
![image](https://user-images.githubusercontent.com/12760278/183043870-f4fa2dc2-c830-4ad5-9202-af278f1a6ca4.png)

/%processing_options/  -> /resize:fill:1024:0/gravity:sm/enlarge:0/unsharpening:always:1:24/
![image](https://user-images.githubusercontent.com/12760278/183044107-aee95b8f-43b5-4b09-8a70-a242e273512d.png)

/%processing_options/  -> /resize:fit:1024:0/gravity:sm/enlarge:0/unsharpening:always:1:24/
![image](https://user-images.githubusercontent.com/12760278/183044287-33baab2b-da75-4a70-818c-83c245149a42.png)

/%processing_options/   -> /resize:fill-down:1024:0/gravity:sm/enlarge:0/unsharpening:always:1:24/ 
![image](https://user-images.githubusercontent.com/12760278/183044369-201da5d5-4227-4c00-91d5-53365fed1026.png)

/%processing_options/  -> /resize:force:1024:0/gravity:sm/enlarge:0/unsharpening:always:1:24/
![image](https://user-images.githubusercontent.com/12760278/183044503-09fbc930-f85c-4ba3-a416-1c22864898ff.png)

/%processing_options/   -> /resize:fill:1024:0/resizing_algorithm:nearest/gravity:sm/enlarge:0/
![image](https://user-images.githubusercontent.com/12760278/183044650-10c64135-9e7c-4d50-870a-2fcc6c9835cc.png)

/%processing_options/ -> /resize:fill:1024:0/resizing_algorithm:linear/gravity:sm/enlarge:0/
![image](https://user-images.githubusercontent.com/12760278/183044703-8be3fcb3-f26e-4311-b3e3-30914619f7a9.png)

/%processing_options/  -> /resize:fill:1024:0/resizing_algorithm:cubic/gravity:sm/enlarge:0/
![image](https://user-images.githubusercontent.com/12760278/183044898-29d49430-2be6-4592-9334-0d66208af978.png)

/%processing_options/  -> /resize:fill:1024:0/resizing_algorithm:lanczos2/gravity:sm/enlarge:0/
![image](https://user-images.githubusercontent.com/12760278/183045021-9e1a968b-d210-4bcf-b443-1c6ec17a8258.png)

/%processing_options/ -> /resize:fill:1024:0/resizing_algorithm:lanczos3/gravity:sm/enlarge:0/
![image](https://user-images.githubusercontent.com/12760278/183045183-e57a2689-1c21-42e0-8926-f218bfc3230d.png)

/%processing_options/ -> /resize:fill:1024:0/gravity:sm/enlarge:0/jpeg_options:0:1:0:0:0:0/
![image](https://user-images.githubusercontent.com/12760278/183045319-a41009ec-3271-4e1d-bce6-0477a4801668.png)

/%processing_options/ -> /resize:fill:1024:0/gravity:sm/enlarge:0/jpeg_options:0:0:0:1:0:0/
![image](https://user-images.githubusercontent.com/12760278/183045439-95a6aed3-777c-4930-8b9a-1d957959744c.png)

**Comment by DarthSim** [comment]
Hi @chaimch!

The font used for the small text is very narrow, the lines are only 1px wide. Because of the font weight, this text is very sensitive to scaling.

You scale from 2400px to 1024px, so your shrink factor is 2400/1024=2,34375 which is not integer. This ain't good too as with fractional shrink factor we have to do the reduce step after the subsampling step.

Looks at these pictures. The text on the first one is sharper than on the second one even though the width of the first one is smaller than the width of the second one. That's because we shrank the first picture exactly twice, and the second picture's shrink factor is fractional.

![picture1](https://user-images.githubusercontent.com/380856/183113801-19267555-bc2e-4b7a-b8ed-1482b339c622.png)
![picture2](https://user-images.githubusercontent.com/380856/183113831-ef59e900-6f42-4c4f-83c1-1415009be280.png)


Long story short, I don't think it's possible to do something for this exact case.
